docs/spark/tutorials/streaming.md

```diff
@@ -0,0 +1,168 @@
+
+2. To start a new connection, open a new console and run the following command which connects to localhost on port 9999.
+
+   On Windows:
+
+   ```console
+   nc -vvv -l -p 9999
+   ```
+
+   On Linux:
+
+   ```console
+   nc -lk 9999
+   ```
+
+   Your Spark program listens for the input you type into this command prompt.
+
+### Create a SparkSession
+
+1. Add the following additional `using` statements to the top of the *Program.cs* file in *mySparkStreamingApp*:
+
+   ```csharp
+   using System;
+   using Microsoft.Spark.Sql;
+   using Microsoft.Spark.Sql.Streaming;
+   using static Microsoft.Spark.Sql.Functions;
+   ```
+
+1. Add the following code to your `Main` method to create a new `SparkSession`. The Spark Session is the entry point to programming Spark with the Dataset and DataFrame API.
+
+   ```csharp
+   SparkSession spark = SparkSession
+        .Builder()
+        .AppName("Streaming example with a UDF")
+        .GetOrCreate();
+   ```
+
+   Calling the *spark* object created above allows you to access Spark and DataFrame functionality throughout your program.
+
+### Connect to a stream with ReadStream()
+
+The `ReadStream()` method returns a `DataStreamReader` that can be used to read streaming data in as a `DataFrame`. Include the host and port information to tell your Spark app where to expect its streaming data.
+
+```csharp
+DataFrame lines = spark
+    .ReadStream()
+    .Format("socket")
+    .Option("host", hostname)
+    .Option("port", port)
+    .Load();
+```
+
+## Register a user-defined function
+
+You can use UDFs, *user-defined functions*, in Spark applications to perform calculations and analysis on your data.
+
+Add the following code to your `Main` method to register a UDF called `udfArray`. 
+
+```csharp
+Func<Column, Column> udfArray =
+    Udf<string, string[]>((str) => new string[] { str, $"{str} {str.Length}" });
+```
+
+This UDF processes each string it receives from the netcat terminal to produce an array that includes the original string (contained in *str*), followed by the original string concatenated with the length of the original string. 
+
+For example, entering *Hello world* in the netcat terminal produces an array where:
+
+* array\[0] = Hello world
+* array\[1] = Hello world 11
+
+## Use SparkSQL
+
+Use SparkSQL to perform various functions on the data stored in your DataFrame. It's common to combine UDFs and SparkSQL to apply a UDF to each row of a DataFrame.
+
+```csharp
+DataFrame arrayDF = lines.Select(Explode(udfArray(lines["value"])));
+```
+
+This code snippet applies *udfArray* to each value in your DataFrame, which represents each string read from your netcat terminal. Apply the SparkSQL method `Explode` to put each entry of your array in its own row. Finally, use `Select` to place the columns you've produced in the new DataFrame *arrayDF.*
```

Comment:
Should `Explode` and `Select` be xrefs?

```markdown
<xref:Microsoft.Spark.Sql.Functions.Explode%2A>
<xref:Microsoft.Spark.Sql.DataFrame.Select%2A>
```

If you think they should be, I'd limit the xrefs to those APIs that are part of Spark for .NET, not all .NET APIs.